### PR TITLE
Fix NPE when no domainConfig is defined in unit test

### PR DESCRIPTION
**Pull request by jshum2479**
The NPE is preventing new code merges https://github.com/oracle/weblogic-kubernetes-operator/pull/2948

### CODE REVIEW

File: operator/src/test/java/oracle/kubernetes/operator/helpers/DomainIntrospectorJobTest.java [hunk 1]
```diff
@@ -601,12 +601,13 @@ void whenJobCreatedWithoutFluentdTerminatedDuringIntropsection() {
...
     testSupport.defineResources(jobPod);
     defineNormalFluentdContainerInIntrospection();
+    testSupport.addToPacket(DOMAIN_TOPOLOGY, createDomainConfig("cluster-1"));
 
     Packet packet = testSupport.runSteps(JobHelper.createIntrospectionStartStep(terminalStep));
     logRecords.clear();
     assertThat(packet.get(JOB_POD_INTROSPECT_CONTAINER_TERMINATED),
             equalTo(JOB_POD_INTROSPECT_CONTAINER_TERMINATED_MARKER));
-    assertThat(terminalStep.wasRun(), is(false));
+    assertThat(terminalStep.wasRun(), is(true));
```
Comment (russgold) [inline operator/src/test/java/oracle/kubernetes/operator/helpers/DomainIntrospectorJobTest.java:610]: Can you clarify the purpose of the test in its name? I think, after speaking with you, that the intent is that JOB_POD_INTROSPECT_CONTAINER_TERMINATED_MARKER is added to the packet, and has nothing to do with the terminalStep. If so, you can remove that last assertion, and change the name to something like 'whenJobCreatedWithoutFluentd_recordTerminatedMarkerInPacket` (if that is what is going on here. I am still not sure of the purpose of this test).

**Review by russgold — CHANGES_REQUESTED** [review]

File: operator/src/test/java/oracle/kubernetes/operator/helpers/DomainIntrospectorJobTest.java (hunk 1 above)
Comment (jshum2479) [reply operator/src/test/java/oracle/kubernetes/operator/helpers/DomainIntrospectorJobTest.java:610]: I changed the name to `whenJobCreatedWithFluentdAndSuccessIntropsection_ensureJobIsTerminatedAndJobTerminatedMarkerInserted` but leave the last assert to make sure the job will run to it's end

Comment (russgold) [reply operator/src/test/java/oracle/kubernetes/operator/helpers/DomainIntrospectorJobTest.java:610]: The purpose of the terminal step is to verify that the make-right runs to the end. Is that the intent?

**Review by russgold — APPROVED** [review]

**Review by doxiao — APPROVED** [review]
LGTM.

**Pull request merged**